message feed: Improve Color Contrast for alert_word Highlighting in Night Mode

amaranand360 opened the pull request:
This pull request addresses various style enhancements in dark mode to improve readability and accessibility:
- All colors used in this pull request have been suggested  in [#12614(comment)](https://github.com/zulip/zulip/issues/12614#issuecomment-1556028086)
- Adjusts the color and background-color of alert words for better visibility.
- Modifies the link color to enhance contrast and readability.
- Improves the search result background-color for a more cohesive dark mode experience.
- I've ensured that the changes made for dark mode don't affect the light mode. Attached is a screenshot of the light mode that shows the adjustments made exclusively for dark mode, ensuring readability and accessibility.


Fixes: #12614 

**Screenshots and screen captures:**

- **Dark Mode Before Changes for Alert Word:**

<img width="1435" alt="Screenshot 2024-02-11 at 2 37 51 PM" src="https://github.com/zulip/zulip/assets/122713145/b38e8a5d-face-40d4-8920-c710b6a279b4">

- **Dark Mode After Changes for Alert Word:**

<img width="1301" alt="Screenshot 2024-02-03 at 12 50 23 PM" src="https://github.com/zulip/zulip/assets/122713145/019835b0-e4f1-4e67-956c-4c4c7bb2987b">

- **Dark Mode Before Changes for search results**

<img width="1433" alt="Screenshot 2024-02-11 at 2 38 32 PM" src="https://github.com/zulip/zulip/assets/122713145/4735bece-dee6-4597-a73f-08f2ef623f2e">

- **Dark Mode After Changes for search results**

<img width="1413" alt="Screenshot 2024-02-03 at 12 54 37 PM" src="https://github.com/zulip/zulip/assets/122713145/630ee44e-d20c-407e-9f25-98ffa73c4acc">

- **Light Mode:**

  - **Doesn't Effect the Changes for Search Results:**

<img width="1406" alt="Screenshot 2024-02-03 at 12 56 50 PM" src="https://github.com/zulip/zulip/assets/122713145/124bc7f7-698f-49c1-9158-f221768e0331">

  - **Doesn't Effect the Changes for Alert Word:**

<img width="1214" alt="Screenshot 2024-02-03 at 12 57 13 PM" src="https://github.com/zulip/zulip/assets/122713145/180fd3da-6372-4e91-b267-fe2af728a01d">

  - **Before Changes for highlighted link text:**

<img width="1434" alt="Screenshot 2024-02-15 at 11 27 46 AM" src="https://github.com/zulip/zulip/assets/122713145/13123d19-6a5c-4552-bfd5-c5e0112e3b78">
  
  - **After Changes for highlighted link text:**
<img width="1433" alt="Screenshot 2024-02-15 at 11 43 05 AM" src="https://github.com/zulip/zulip/assets/122713145/bbd5741c-407c-4ce0-8450-f3c8a74d9f8c">






<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

LaPulgaaa commented:
Thanks for the contribution @amaranand360 !!. The maintainers will review your work shortly, but I wanted to state that your commit message and PR discription don't match Zulip's style. 
You can refer to [Zulip commit message guidelines](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-messages)  for details.It will save some review time.

amaranand360 commented:
@LaPulgaaa  I created a pull request for issue #12614  three days ago but I haven't received any updates from the reviewer. Please let me know if there's anything I need to do on my end.

amaranand360 commented:
@LaPulgaaa I have updated all commits of PR to match Zulip's style. Please let me know if there is anything else needed from my side.

LaPulgaaa commented:
This still might not be fully correct. Plus you may also want to update the PR title. Please go through the docs I linked above .

amaranand360 commented:
@LaPulgaaa I have updated the PR title and refined the commit message by removing generic terms like 'fix.' and let me know if there are any additional changes or actions required on my end

alya commented:
Thanks! Does this implement the design changes described in https://github.com/zulip/zulip/issues/12614#issuecomment-1556028086?

alya commented:
If so, this should be explicitly noted in the PR description. If not, please update this PR to do so, since that's the current next step for this issue.

amaranand360 commented:
@alya Yes, it does. I have Used the suggested color design changes described in [#12614(comment)](https://github.com/zulip/zulip/issues/12614#issuecomment-1556028086). The originally provided color values were in hexadecimal format, but I have converted them to HSL format, which aligns with the color format used in Zulip.

amaranand360 commented:
@LaPulgaaa  and @alya , I’ve changed the description of my Pull Request based on your suggestions.  Please take a moment to review, and if there are any additional improvements  let me know. Thank you !

alya commented:
Thanks! The PR description is still not very clear. Were light theme colors changed?

amaranand360 commented:
@alya No, the light theme colors remain unchanged. I've updated the pull request description accordingly and attached screenshots to confirm. Please review the updated information. Thanks!

alya commented:
Thanks! Could you please update your commit message to match the [commit style guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages)?

In general, making an effort to communicate as clearly as possible on your PR will make the review process go a lot quicker. :)

amaranand360 commented:
@alya Thank you for your insight, I have updated my PR according to the commit style guidelines. Please review the updated information.

karlstolley commented on web/styles/dark_theme.css:
```diff
@@ -1081,7 +1081,8 @@
     /* Search highlight used in both topics and rendered_markdown */
     .highlight {
-        background-color: hsl(51deg 100% 23%);
+        background-color: hsl(51deg 100% 72%);
```
Let's get all of these colors into CSS variables in `zulip.css`. Your PR is a great opportunity to just completely remove these lines from `dark_theme.css` entirely.

karlstolley replied:
And especially once this removal is in place, make the `.highlight`-related work its own commit.

amaranand360 replied:
sure,I will introduce CSS variables for the colors in zulip.css. Moreover, I'll create a separate commit for the .highlight-related changes. can you please elaborate on what do you mean by completely remove these lines

karlstolley commented on web/styles/rendered_markdown.css:
```diff
@@ -210,8 +210,15 @@
 
+    a .alert-word {
+        background-color: var(--color-background-alert-word);
+        color: hsl(210deg 100% 25%);
```
Let's make a variable for the `color` values too. Also, I don't understand how this doesn't change light mode? Your screenshots suggest that light mode isn't affected.

amaranand360 replied:
@karlstolley Thank you for highlighting that. I've resolved the issue by making the required adjustments. The color modifications are now specifically tailored to dark mode, and I have included screenshots to confirm that light mode remains unchanged.

amaranand360 commented:
@karlstolley I have implemented the changes as per your suggestions. Kindly review the updated information and provide your feedback.

karlstolley commented:
You need to clean up your commit history with `git rebase`, and not just stack new commits onto the old ones.

amaranand360 commented:
I am

amaranand360 commented:
Hi @karlstolley ,

I've opened a new PR, #28973, to address Issue #12614. Following your suggestion, I've cleaned up the commit history and introduced CSS variables. Could you please review the changes?

I apologize for closing this pull request due to merge conflicts.

alya commented:
You shouldn't need to start a new pull request; you can reopen this one and push your changes here. This will help by keeping all the review comments in one place.

alya commented:
Oh, it looks like you deleted the branch. OK, request for next time then.
